### ISSUE TITLE
Device calendar plugin ran into an issue. Platform specific exception [500], with message :"That account does not allow calendars to be added or removed.", has been thrown.

### ISSUE THREAD

**Issue by arsalalgorepublic** [builttoroam/device_calendar#420]
I'm not able to create calendar on iPhone 12 pro max using  ios version 15.4.1

Error code: 599
Error Message: Device calendar plugin ran into an issue. Platform specific exception [500], with message :"That account does not allow calendars to be added or removed.", has been thrown.

Did anyone face this issue?

**Comment by thomassth** [comment]
Does said account allow calendars to be created or deleted?

We may need more information on the account info side

**Comment by arsalalgorepublic** [comment]
Actually, the problem is that the user has set his default calender as a google calendar on an IOS device. It works when we set the default calender as an iCloud calender.

**Comment by thomassth** [comment]
It sounds like a problem on Google calendar's setup, if icloud works fine

**Comment by arsalalgorepublic** [comment]
Yes, It seems like Google Calendar doesn't allow us to read and write on iOS even though I checked on android it's working fine with google calendar as well.

**Comment by stephenjen** [comment]
For iOS, the resolution seems to be to always create new calendars in iCloud?  Where to create new calendars should be separate to where the default calendar resides.

**Comment by arsalalgorepublic** [comment]
> For iOS, the resolution seems to be to always create new calendars in iCloud? Where to create new calendars should be separate to where the default calendar resides.

Exactly I'm creating a new calendar but we have no such option to create a calendar in  iCloud . In Ios, it chooses to create a calendar on the default selected calender in my case it is google calender so I ran into an error.

**Comment by thomassth** [comment]
Looking at how other framework did things, it seems that no one can add calendar to specific calendar accounts?
Maybe this is another ecosystem lock in
https://github.com/EddyVerbruggen/Calendar-PhoneGap-Plugin/blob/81ac2df9d99d23d68b25120654bb792bd8f49e42/src/ios/Calendar.m#L935

**Comment by ebsangam** [comment]
On iOS Calendar app make sure Gmail Calendar is connected properly and re-authenticate if not.

**Comment by ConnorDykes** [comment]
I am still seeing this issue is there a work around?
